qa.hawk/src/main/java/org/aksw/hawk/nlp/SentenceToSequence.java

```diff
@@ -50,60 +55,263 @@ public static void runPhraseCombination(final HAWKQuestion q, final List<String>
 			// look for start "RB|JJ|NN(.)*"
 			//add JJS to the list AND add a new condition to prevent addding many|much|old
 			if (subsequence.isEmpty() && (null != pos) && (( pos.matches("CD|NN(.)*|RB(.)*")) || (pos.matches("JJ(.)*") && !token.matches("many|much|old") )  )) {
-				subsequence.add(token);
+				tempSubsequance2 = tempSubsequance2 + " " + token;
+				tempSubsequance2 = tempSubsequance2.trim();
 			}
 			
 			else if (!subsequence.isEmpty() && (null != pos) && lastPos.matches("JJ(.)*|HYPH|NN(.)*") && pos.matches("VB(.)*|IN|WDT") && (null != nextPos) && nextPos.matches("NN(.)*") )
 			{
-
-				subsequence.add(token);
+				tempSubsequance2 = tempSubsequance2 + " " + token;
+				tempSubsequance2 = tempSubsequance2.trim();
 			}
 			// split "of the" or "of all" or "against" via pos_i=IN and
 			// pos_i+1=DT
 			else if (!subsequence.isEmpty() && (null != pos) && ((tcounter + 1) < question.size()) && (null != nextPos) && pos.matches("IN") && !token.matches("of")
 			        && nextPos.matches("(W)?DT|NNP(S)?")) {
-				if (subsequence.size() > 1) {
-					transformTree(subsequence, q, tokenOffset);
-				}
-				subsequence = Lists.newArrayList();
+				           namedEntityOrCombinedPhrase = checkNamedEntites(tempSubsequance2,q);
```

Comment:
I think that this block of code is exactly the same as the one below, could you extract a function?